src/transformers/models/maskformer/feature_extraction_maskformer.py

```diff
@@ -0,0 +1,549 @@
+
+                Returns:
+                    Tensor: A tensor of shape `batch_size, num_labels, height, width`
+        """
+        # class_queries_logitss has shape [BATCH, QUERIES, CLASSES + 1]
+        class_queries_logits = outputs.class_queries_logits
+        # masks_queries_logits has shape [BATCH, QUERIES, HEIGHT, WIDTH]
+        masks_queries_logits = outputs.masks_queries_logits
+        if target_size is not None:
+            masks_queries_logits: Tensor = interpolate(
+                masks_queries_logits,
+                size=target_size,
+                mode="bilinear",
+                align_corners=False,
+            )
+        # remove the null class `[..., :-1]`
+        masks_classes: Tensor = class_queries_logits.softmax(dim=-1)[..., :-1]
+        # mask probs has shape [BATCH, QUERIES, HEIGHT, WIDTH]
+        masks_probs: Tensor = masks_queries_logits.sigmoid()
+        # now we want to sum over the queries,
+        # $ out_{c,h,w} =  \sum_q p_{q,c} * m_{q,h,w} $
+        # where $ softmax(p) \in R^{q, c} $ is the mask classes
+        # and $ sigmoid(m) \in R^{q, h, w}$ is the mask probabilities
+        # b(atch)q(uery)c(lasses), b(atch)q(uery)h(eight)w(idth)
+        segmentation: Tensor = torch.einsum("bqc, bqhw -> bchw", masks_classes, masks_probs)
+
+        return segmentation
+
+    def remove_low_and_no_objects(
+        self, masks: Tensor, scores: Tensor, labels: Tensor, object_mask_threshold: float, num_labels: int
+    ) -> Tuple[Tensor, Tensor, Tensor]:
+        """
+
+        Binarize the given masks using `object_mask_threshold`, it returns the associated values of `masks`, `scores`
+        and `labels`
+
+        Args:
+            masks (Tensor): A tensor of shape `(num_queries, height, width)`
+            scores (Tensor): A tensor of shape `(num_queries)`
+            labels (Tensor): A tensor of shape `(num_queries)`
+            object_mask_threshold (float): A number between 0 and 1 used to binarize the masks
+
+        Raises:
+            ValueError: When the first dimension doesn't match in all input tensors
+
+        Returns:
+            Tuple[Tensor, Tensor, Tensor]: The inputs tensors without the region with `< object_mask_threshold`
+        """
+        if not (masks.shape[0] == scores.shape[0] == labels.shape[0]):
+            raise ValueError("mask, scores and labels must have the same shape!")
+
+        to_keep: Tensor = labels.ne(num_labels) & (scores > object_mask_threshold)
+
+        return masks[to_keep], scores[to_keep], labels[to_keep]
+
+    def post_process_semantic_segmentation(
+        self, outputs: MaskFormerForInstanceSegmentationOutput, target_size: Tuple[int, int] = None
+    ) -> Tensor:
+        """Converts the output of [`MaskFormerForInstanceSegmentationOutput`] into semantic segmentation predictions. Only
+        supports PyTorch.
+
+                Args:
+                    outputs (MaskFormerForInstanceSegmentationOutput): The outputs from MaskFor
```

Comment:
Same comment as above, properly link the class and write a description.

Reply:
added the "`"